sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/customtokenproxy/CustomTokenProxyHttpClient.java

```diff
@@ -0,0 +1,322 @@
+                    }
+                    LOGGER.warning("CA file " + caFile + " is empty, using cached SSL context from previous load");
+                    return cachedSSLContext;
+                }
+
+                if (cachedSSLContext == null
+                    || currentLength != cachedFileContentLength
+                    || !Arrays.equals(currentHash, cachedFileContentHash)) {
+                    cachedSSLContext = createSslContextFromBytes(currentContent);
+                    cachedFileContentLength = currentLength;
+                    cachedFileContentHash = currentHash;
+                    cachedSslSocketFactory = null;
+                }
+            }
+            return cachedSSLContext;
+
+        } catch (Exception e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException("Failed to initialize SSLContext for proxy", e));
+        }
+    }
+
+    // Create SSLContext from byte array containing PEM certificate data
+    private SSLContext createSslContextFromBytes(byte[] certificateData) {
+        try (InputStream inputStream = new ByteArrayInputStream(certificateData)) {
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+
+            List<X509Certificate> certificates = new ArrayList<>();
+            while (true) {
+                try {
+                    X509Certificate cert = (X509Certificate) cf.generateCertificate(inputStream);
+                    certificates.add(cert);
+                } catch (CertificateException e) {
+                    break;
+                }
+            }
+
+            if (certificates.isEmpty()) {
+                throw LOGGER.logExceptionAsError(new IllegalStateException("No valid certificates found"));
+            }
+            return createSslContext(certificates);
+        } catch (Exception e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException("Failed to create SSLContext from bytes", e));
+        }
+    }
+
+    private SSLContext createSslContext(List<X509Certificate> certificates) {
+        try {
+            KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keystore.load(null, null);
+            int index = 1;
+            for (X509Certificate caCert : certificates) {
+                keystore.setCertificateEntry("ca-cert-" + index, caCert);
+                index++;
+            }
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(keystore);
+
+            SSLContext context = SSLContext.getInstance("TLS");
+            context.init(null, tmf.getTrustManagers(), null);
+            return context;
+        } catch (Exception e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException("Failed to create SSLContext", e));
+        }
+    }
+
+    private static HostnameVerifier sniAwareVerifier(String sniName, URL customProxyUrl) {
```

Comment:
currently I think we skip the default Verifier checks, when we override this. 
Try adding this

```java
private static HostnameVerifier sniAwareVerifier(String sniName, URL customProxyUrl) {
    HostnameVerifier defaultVerifier = HttpsURLConnection.getDefaultHostnameVerifier();

    return (hostname, session) -> {
        boolean defaultPass = defaultVerifier.verify(hostname, session);

        if (!defaultPass) {
            return false;
        }
        
         //rest of logic
         
```